static/js/ducks/spectra.js

```diff
@@ -28,29 +31,33 @@ export function parseASCIISpectrum(data) {
   );
 }
 
+export function deleteSpectrum(id) {
+  return API.DELETE(`/api/spectrum/${id}`, DELETE_SPECTRUM);
+}
+
 export function uploadASCIISpectrum(data) {
   return API.POST(`/api/spectrum/ascii`, UPLOAD_SPECTRUM, data);
 }
 
 // Websocket message handler
-messageHandler.add((actionType, payload, dispatch) => {
-  if (actionType === FETCH_SOURCE_SPECTRA) {
-    dispatch(fetchSourceSpectra(payload.obj_id));
+messageHandler.add((actionType, payload, dispatch, getState) => {
+  if (actionType === REFRESH_SOURCE_SPECTRA) {
+    const state = getState().spectra;
+    if (Object.keys(state).includes(payload.obj_id)) {
+      dispatch(fetchSourceSpectra(payload.obj_id));
+    }
   }
 });
 
 const reducer = (state = { parsed: null }, action) => {
   switch (action.type) {
     case FETCH_SOURCE_SPECTRA_OK: {
-      const spectra = action.data;
-      if (spectra.length > 0) {
-        const sourceID = spectra[0].obj_id;
-        return {
-          ...state,
-          [sourceID]: spectra,
-        };
-      }
-      return state;
+      const payload = action.data;
+      const sourceID = payload.obj_id;
+      return {
+        ...state,
+        [sourceID]: payload.spectra,
```

Comment:
Rather than always keep past-viewed spectra in the state tree, why don't we just overwrite with the new, relevant data? With this implementation, if a user views the spectra associated with many sources in a given browser session, all of that data will persist in the redux state, even though they're no longer relevant. This will cause excessive re-fetches as per the logic in the WS message handler above, because all past source spectra sticks around in the state.

Reply:
Yeah good point.

Reply:
OK, just pushed up a fix for this, let me know what you think @acrellin.